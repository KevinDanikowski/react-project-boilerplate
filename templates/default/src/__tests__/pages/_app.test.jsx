import React from 'react'
import App from '../../pages/_app'

App.displayName = 'App'

jest.mock('../../utils/apollo', () => ({
  withApollo: () => async Component => {
    const props = await Component.getInitialProps({ Component })

    // eslint-disable-next-line
    return <Component {...props} />
  }
}))

const mockComponent = () => {
  return <div>Ok!</div>
}

describe('App', () => {
  it('controls the root NextJS app', () => {
    expect(<App Component={mockComponent} pageProps={{}} />).toMatchSnapshot()
  })
})
