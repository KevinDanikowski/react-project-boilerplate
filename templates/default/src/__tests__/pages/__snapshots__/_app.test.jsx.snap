// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App controls the root NextJS app 1`] = `
<App
  Component={[Function]}
  pageProps={Object {}}
/>
`;
